src/bootstrap.ts

```diff
@@ -0,0 +1,400 @@
+    * @param {string} token - Authentication Token.
+    *
+    */
+    const options: RequestInit = {
+      method: 'PATCH',
+      headers: {
+        'Content-Type': this.contentType,
+        Authorization: `Bearer ${token}`
+      },
+      body: JSON.stringify({ client_cert: clientCert, client_key: clientKey, ca_cert: caCert })
+    }
+    try {
+      const response = await fetch(
+        new URL(`things/configs/certs/${thingId}`, this.bootstrapsUrl).toString(), options
+      )
+      if (!response.ok) {
+        const errorRes = await response.json()
+        throw this.bootstrapError.HandleError(errorRes.error, response.status)
+      }
+      const updatedBootstrap: Bootstrap = await response.json()
+      return updatedBootstrap
+    } catch (error) {
+      throw error
+    }
+  }
+
+  public async RemoveBootstrap (thingId: string, token: string): Promise<Response> {
+    // Remove a bootstrap configuration
+    /**
+    * @method Remove - Allows a logged in user to delete a bootstrap configuration.
+    * @param {string} config_id - The config_id of the configuration to be deleted.
+    * This can also mean the thing_id.
+    * @param {string} token - Authentication Token.
+    */
+    const options: RequestInit = {
+      method: 'DELETE',
+      headers: {
+        'Content-Type': this.contentType,
+        Authorization: `Bearer ${token}`
+      }
+    }
+    try {
+      const response = await fetch(
+        new URL(`things/configs/${thingId}`, this.bootstrapsUrl).toString(), options
+      )
+      if (!response.ok) {
+        const errorRes = await response.json()
+        throw this.bootstrapError.HandleError(errorRes.error, response.status)
+      }
+      const removeResponse: Response = { status: response.status, message: 'Configuration Removed' }
+      return removeResponse
+    } catch (error) {
+      throw error
+    }
+  }
+
+  public async BootstrapSecure (externalId: string, externalKey: string, cryptoKey: string): Promise<Bootstrap> {
+    // Secures a Bootstrap configuration
+    /**
+    * @method Bootstrap - Retrieves a configuration with given external ID and external key.
+    * @param {string} external_id - The external Id of the configuration to be retrieved.
+    * @param {string} external_key - The encrypted external key of the configuration to be retrieved.
+    * @return {object} - Returns a secured Bootstrap.
+    */
+
+    const options: RequestInit = {
+      method: 'GET',
+      headers: {
+        'Content-Type': this.contentType,
+        Authorization: `Thing ${externalKey}`
+      }
+    }
+    try {
+      const response = await fetch(
+        new URL(`things/bootstrap/secure/${externalId}`, this.bootstrapsUrl).toString(), options
+      )
+      if (!response.ok) {
+        const errorRes = await response.json()
+        throw this.bootstrapError.HandleError(errorRes.error, response.status)
+      }
+      const securedBootstrap: Bootstrap = await response.json()
```

Comment:
You need to decrypt the response before returning it back to the user. Again check the `go sdk`